drv/transceivers-server/src/udp.rs

```diff
@@ -10,25 +10,28 @@
 //!
 //! All of the API types in `transceiver_messages` operate on **physical**
 //! ports, i.e. an FPGA paired by a physical port index (or mask).
+//!
+use counters::Count;
+use ringbuf::*;
+use userlib::UnwrapLite;
+
 use crate::{FrontIOStatus, ServerImpl};
 use drv_sidecar_front_io::transceivers::{
     FpgaI2CFailure, LogicalPort, LogicalPortFailureTypes, LogicalPortMask,
     ModuleResult, ModuleResultNoFailure, ModuleResultSlim, PortI2CStatus,
 };
 use hubpack::SerializedSize;
-use ringbuf::*;
 use task_net_api::*;
 use transceiver_messages::{
     mac::MacAddrs,
     message::*,
     mgmt::{ManagementInterface, MemoryRead, MemoryWrite, Page},
     ModuleId,
 };
-use userlib::UnwrapLite;
 
 ////////////////////////////////////////////////////////////////////////////////
 
-#[derive(Copy, Clone, PartialEq)]
+#[derive(Copy, Clone, PartialEq, Count)]
 enum Trace {
     None,
```

Comment:
in general, it's good to add `#[count(skip)]` to the `None`/`Empty` entries --- since we will never actually record them, we don't need to generate a counter for them.
```suggestion
    #[count(skip)]
    None,
```